{
  "comments": [
    {
      "key": {
        "uuid": "1abeadc6_309ea8b9",
        "filename": "nodepool/nodepool.py",
        "patchSetId": 1
      },
      "lineNbr": 1512,
      "author": {
        "id": 1
      },
      "writtenOn": "2014-08-23T15:25:38Z",
      "side": 0,
      "message": "this section should be fine",
      "revId": "a564cbc57f4863cd063f52cdc321a0e8b39cc80c",
      "serverId": "4a232e18-c5a9-48ee-94c0-e04e7cca6543",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "1abeadc6_15719e0b",
        "filename": "nodepool/nodepool.py",
        "patchSetId": 1
      },
      "lineNbr": 1720,
      "author": {
        "id": 5263
      },
      "writtenOn": "2014-08-23T14:44:51Z",
      "side": 0,
      "message": "This was causing scheduled image updates to fail with:\n\n    ERROR nodepool.NodePool: Exception in periodic image update:\n    Traceback (most recent call last):\n      File \"/usr/local/lib/python2.7/dist-packages/nodepool/nodepool.py\", line 1690, in _doUpdateImages\n        self.updateImages(session)\n      File \"/usr/local/lib/python2.7/dist-packages/nodepool/nodepool.py\", line 1720, in updateImages\n        self.updateImage(session, provider.name, label.image)\n    AttributeError: \u0027str\u0027 object has no attribute \u0027name\u0027",
      "revId": "a564cbc57f4863cd063f52cdc321a0e8b39cc80c",
      "serverId": "4a232e18-c5a9-48ee-94c0-e04e7cca6543",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "1abeadc6_70a4a007",
        "filename": "nodepool/nodepool.py",
        "patchSetId": 1
      },
      "lineNbr": 1720,
      "author": {
        "id": 1
      },
      "writtenOn": "2014-08-23T15:25:38Z",
      "side": 0,
      "message": "I think it might be more idiomatic with the rest of nodepool if we reversed the fix.  If we changed \u0027label.providers\u0027 to \u0027label.providers.values()\u0027, it should work.",
      "parentUuid": "1abeadc6_15719e0b",
      "revId": "a564cbc57f4863cd063f52cdc321a0e8b39cc80c",
      "serverId": "4a232e18-c5a9-48ee-94c0-e04e7cca6543",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "1abeadc6_105c4c18",
        "filename": "nodepool/nodepool.py",
        "patchSetId": 1
      },
      "lineNbr": 2177,
      "author": {
        "id": 1
      },
      "writtenOn": "2014-08-23T15:25:38Z",
      "side": 0,
      "message": "this should be fine actually",
      "revId": "a564cbc57f4863cd063f52cdc321a0e8b39cc80c",
      "serverId": "4a232e18-c5a9-48ee-94c0-e04e7cca6543",
      "unresolved": false
    }
  ]
}